Divide backplane rbac into directories based on roles, use aggregated roles for cssre add-ons

karthikperu7 opened the pull request:
## Summary of changes:

* moved files to their own directories according to the teams/user groups
* divided cssre into rhoam and rhmi
  * This is the same change that was made recently under layered-sre-authorization for rhmi and rhoam. The only difference is that this does not duplicate the aggregated role. Once backplane RBAC is moved to production ( see deploy/backplane/config.yaml) we should move the cssre aggregated cluster role to deploy/backplane/layered-sre/rh(mi|oam)
* Added a basic README on the top backplane dir, can be extended later

### Jira
[OSD-5999](https://issues.redhat.com/browse/OSD-5999)

karthikperu7 commented:
/hold

for discussion/review

karthikperu7 commented:
cc: @jewzaam @wanghaoran1988

jewzaam reviewed (changes requested):
The "cluster-admin" didn't get moved correctly.  ClusterRoleBinding file is a namespace.  Missing the actual ClusterRoleBinding.

Missing the "aggregate" clusterroles from layered-sre-authorization/{rhmi,rhoam}
With the file names being different it is going to be hard to keep this in sync.  Recommend seeing if a symlink will work so we have a single source of truth.  Or create a script to sync the RBAC as it changes between these.  Prefer we have backplane be the source of truth and the syncing be done as part of `make`, verified by PR check.

karthikperu7 commented:
> The "cluster-admin" didn't get moved correctly. ClusterRoleBinding file is a namespace. Missing the actual ClusterRoleBinding.

I will fix this.
> 
> Missing the "aggregate" clusterroles from layered-sre-authorization/{rhmi,rhoam}

I left out aggregate roles deliberately to avoid duplication between the two directories (backplane and layered-sre-auth) and since backplane can aggregate those roles into backplane/layered-sre/10-cssre-admins-cluster.ClusterRole.yml .

> With the file names being different it is going to be hard to keep this in sync. Recommend seeing if a symlink will work so we have a single source of truth. Or create a script to sync the RBAC as it changes between these. Prefer we have backplane be the source of truth and the syncing be done as part of `make`, verified by PR check.

This is a good point, the files names in the layered-auth are not the most consistent so I wanted to fix that in backplane... I will rename and match to what already is present and also try symlinks. (I have a feeling make targets and PR checks will take me a while :smile:   )

karthikperu7 commented:
> Prefer we have backplane be the source of truth 
Since backplane rbac is not deployed to prod yet, we cannot let backplane have the source of truth at the moment. We can do that after backplane goes to prod.

karthikperu7 commented:
### Next set of changes:

* Fix backplane/elevated-sre/cluster-admin RBAC files
  * fix namespace file name and add the missing cluster-admin rolebinding
* Move common ClusterRole and config files from layered-authorization to backplane. Replace the original files with symlinks pointing to corresponding backplane files (symlinks work!)
* Update layered-sre-authorization/README.md to mention about the files with symlinks

@jewzaam PTAL when you can.

jewzaam commented on deploy/backplane/layered-sre/02-cs-sre-admins.ClusterRoleBinding.yml:
```diff
@@ -1,11 +1,11 @@
 kind: ClusterRoleBinding
 metadata:
-  name: backplane-csssre-admins
+  name: backplane-cs-ssre-admins
```
```suggestion
  name: backplane-cs-sre-admins
```

jewzaam commented on deploy/backplane/layered-sre/rhmi/config.yaml:
```diff
@@ -1,6 +1,4 @@
-      api.openshift.com/addon-rhmi-operator: "true"
-      api.openshift.com/addon-rhmi-operator-internal: "true"
-  matchLabelsApplyMode: "OR"
+      api.openshift.com/addon-managed-api-service: "true"
```
This was changed to the rhoam config, which is wrong.

jewzaam reviewed (changes requested):
There are some CSSRE resources prefixed with backplane, some layered-sre... we need to pick one.  Given the change to symlinks and backplane dir being the source I'd prefer we use backplane.  Probably less churn to do it once (now).

/hold

cc @alanmoran

karthikperu7 commented:
### Summary of next set of changes

* changed names of roles sourced from backplane
  * under `rhmi/layered-cs-sre` -> `backplane-rhmi-cs-sre`
  * under `rhoam/rhoam-cs-sre` -> `backplane-rhoma-cs-sre`
* Changed file names both `backplane` and `layered-sre-authorization`  dirs
  * under rhmi: `layered-cs-sre` -> `rhmi-cs-sre`
* Fixed `rhmi/config.yaml` back to rhmi selector labels
* Fixed typo in `deploy/backplane/layered-sre/02-cs-sre-admins.ClusterRoleBinding.yml`
  * `cs-ssre` -> `cs-sre`
* updated symlinks under `layered-sre-authorization` wrt the file name changes
* Updated the list of symlinks in `layered-sre-authorization/README.md`

karthikperu7 commented:
@jewzaam Can you take a look please when you have some time. :bow:

karthikperu7 commented:
Merged Naveen's recommended changes for layered-sre vs cssre name change standardisation.
Rebased branch.

wanghaoran1988 commented:
> #### Concern
> * Cannot update `config.yaml` files under backplane/cssre as they are symlinked to `config.yaml` files under layered-sre-authorization. Do we remove the syhmlinks and duplicate or let backplane cssre file deploy to production? @jewzaam

To make sure backplane is not on production totally, I guess we need duplicate that one?

karthikperu7 commented:
#### More changes
* Duplicate `config.yaml` under `layered-sre-authorization` and remove the symlink to `backplane/cssre/**config.yaml`
* Add non-prod match expression for SSS under `backplane/cssre/**`
* Update `layered-sre-authorization/README.md` according to the symlink changes

@jewzaam Can you PTAL? We ideally want to merge this before the change freeze starting `2020-12-16`

jewzaam commented:
/lgtm
/approve

karthikperu7 commented:
/unhold
